sklearn/utils/extmath.py

```diff
@@ -188,7 +189,7 @@ def safe_sparse_dot(a, b, *, dense_output=False):
             ret = a_2d @ b
             ret = ret.reshape(*a.shape[:-1], b.shape[1])
         else:
-            ret = np.dot(a, b)
```

Comment:
This change is probably causing the regression in the coordinate descent solver. I will investigate later.